src/KingSystem/Ecosystem/ecosystem.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include <basis/seadTypes.h>
+#include <heap/seadDisposer.h>
+#include <heap/seadExpHeap.h>
+#include <math/seadMathCalcCommon.h>
+#include "KingSystem/Ecosystem/ecoLevelSensor.h"
+#include "KingSystem/Utils/Byaml/Byaml.h"
+
+namespace ksys::eco {
+
+struct EcoMapHeader {
+    u32 unknown;
+    s32 numRows;
+    s32 divisor;
+};
+struct Segment {
+    s16 value;
+    s16 length;
+};
+
+class EcoMapInfo {
+public:
+    EcoMapHeader* header;
+    u32* rowOffsets;
+    char* rows;
+};
+
+enum ActorType {
+
+};
+
+static const char* sActorTypes[] = {
+
+};
+
+struct ActorSpawnInfo;
+
+union StatusEffectVal {
+    f32 _f32;
+    s32 _s32;
+};
+
+struct StatusEffectInfo {
+    bool ng;
+    u8 reserved[3];
+    StatusEffectVal val;
+};
+
+enum StatusEffect {
+    StatusEffect_LifeRecover = 0x0,
+    StatusEffect_LifeMaxUp = 0x1,
+    StatusEffect_GutsPerformance = 0x2,
+    StatusEffect_ResistHot = 0x3,
+    StatusEffect_ResistCold = 0x4,
+    StatusEffect_ResistElectric = 0x5,
+    StatusEffect_MovingSpeed = 0x6,
+    StatusEffect_SwimingSpeed = 0x7,
+    StatusEffect_SwimingAnimRate = 0x8,
+    StatusEffect_ClimbingSpeed = 0x9,
+    StatusEffect_AttackUp = 0xA,
+    StatusEffect_DefenseUp = 0xB,
+    StatusEffect_Quietness = 0xC,
+    StatusEffect_DesertMovingSpeed = 0xD,
+    StatusEffect_ThrowingPower = 0xE,
+    StatusEffect_SnowMovingSpeed = 0xF,
+    StatusEffect_GutsRecoverSpeed = 0x10,
+    StatusEffect_ResistThunder = 0x11,
+    StatusEffect_ArmorChargeAttackAddLevel = 0x12,
+    StatusEffect_ReduceAncientEnemyDamge = 0x13,
+    StatusEffect_MaterSwordAttackUp = 0x14,
+};
+
+static const char* sStatusEffectNames[22] = {"StatusEffect",
```

Comment:
This is dangerous in a header... Every translation unit that includes this header will get its own copy of the array. This should most likely go into the .cpp.